apis/python/src/tiledbsoma/util_iter.py

```diff
@@ -0,0 +1,119 @@
+import abc
+from typing import TypeVar
+
+import numpy as np
+import pyarrow as pa
+import scipy.sparse as sp
+
+# This package's pybind11 code
+import tiledbsoma.libtiledbsoma as clib
+
+from . import somacore  # to be replaced by somacore package, when available
+from .types import NTuple
+
+
+class TableReadIter(somacore.ReadIter[pa.Table]):
+    """Iterator over Arrow Table elements"""
+
+    def __init__(self, sr: clib.SOMAReader):
+        self.sr = sr
+
+    def __next__(self) -> pa.Table:
+        arrow_table = self.sr.read_next()
+        if arrow_table is None:
+            raise StopIteration
+
+        return arrow_table
+
+    def concat(self) -> pa.Table:
+        """Concatenate remainder of iterator, and return as a single Arrow Table"""
+        return pa.concat_tables(self)
+
+    def close(self) -> None:
```

Comment:
I imagine you have good reason for including a stub `close` but not a stub `open` -- I don't understand though why we have one and not the other

Reply:
It is a context manager cleanup protocol, not an array/reader close - see somabase.  There will never be an open in this class - when you get one of these classes, it is a handle to an already open object.